### PR TITLE
KFRD-14 First part of integration tests was added

### PR DESCRIPTION
**Pull request by MedvedevBW**

### CODE REVIEW

File: src/it/scala/com/bwsw/kafka/reader/KafkaReaderIntegrationTest.scala [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+        Try(producer.close())
+        x
+      case Failure(e: Throwable) =>
+        producer.close()
+        throw e
+    }
+  }
+
+  "MockEventHandler" should "handle all events from single Kafka topic and then handle nil events" in { fixture =>
+    val groupId = "group1"
```
Comment (placccebo) [inline src/it/scala/com/bwsw/kafka/reader/KafkaReaderIntegrationTest.scala]: should be a global variable

File: src/it/scala/com/bwsw/kafka/reader/KafkaReaderIntegrationTest.scala [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+      kafkaEndpoints,
+      consumerGroupId,
+      topicInfoList,
+      retrieveCount
+    )
+
+    firstTestEntities.checkpointInfoProcessor.load()
+
+
+    val outputEnvelopes = firstTestEntities.eventHandler.handle(new AtomicBoolean(true))
```
Comment (placccebo) [inline src/it/scala/com/bwsw/kafka/reader/KafkaReaderIntegrationTest.scala]: declare as global (dummyFlag = new AtomicBoolean(true))

File: src/it/scala/com/bwsw/kafka/reader/KafkaReaderIntegrationTest.scala [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    testForEmptyTopics(fixture.kafkaEndpoints, groupId, topicInfoList)
+  }
+
+  private def testForNonEmptyTopics(kafkaEndpoints: String,
+                                    consumerGroupId: String,
+                                    topicInfoList: TopicInfoList,
+                                    retrieveCount: Int,
+                                    expectedTestDataSet: Set[String]): Unit = {
+
+    val firstTestEntities = createTestEntities[String, String, String](
```
Comment (placccebo) [inline src/it/scala/com/bwsw/kafka/reader/KafkaReaderIntegrationTest.scala]: is it "first" about what? the same question is about "second"

File: src/it/scala/com/bwsw/kafka/reader/KafkaReaderIntegrationTest.scala [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+  }
+
+  private def testForEmptyTopics(kafkaEndpoints: String,
+                                 consumerGroupId: String,
+                                 topicInfoList: TopicInfoList): Unit = {
+    val secondTestEntities = createTestEntities[String, String, String](
+      kafkaEndpoints,
+      consumerGroupId,
+      topicInfoList,
+      10
```
Comment (placccebo) [inline src/it/scala/com/bwsw/kafka/reader/KafkaReaderIntegrationTest.scala]: magic number))

File: src/it/scala/com/bwsw/kafka/reader/KafkaReaderIntegrationTest.scala [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+      case (topicData, value) => new ProducerRecord[String, String](topicData, 0, "key", value)
+    }
+
+    fixture.producer.send(producerRecords)
+
+    testForNonEmptyTopics(
+      fixture.kafkaEndpoints,
+      groupId,
+      topicInfoList,
+      countOfTestData/2 * 2,
```
Comment (placccebo) [inline src/it/scala/com/bwsw/kafka/reader/KafkaReaderIntegrationTest.scala]: what is the point?

**Comment by MedvedevBW** [comment]
[run tests]

File: src/it/scala/com/bwsw/kafka/reader/KafkaReaderIntegrationTest.scala [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+      case Success(x) =>
+        Try(producer.close())
+        x
+      case Failure(e: Throwable) =>
+        producer.close()
+        throw e
+    }
+  }
+
+  "MockEventHandler" should "handle all events from single Kafka topic and then handle nil events" in { fixture =>
```
Comment (placccebo) [inline src/it/scala/com/bwsw/kafka/reader/KafkaReaderIntegrationTest.scala]: it should "process all events from single Kafka topic, save checkpoint data (indicating what events have been processed), run from a scratch and not receive any message"

File: src/it/scala/com/bwsw/kafka/reader/KafkaReaderIntegrationTest.scala [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+      groupId,
+      topicInfoList,
+      countOfTestData,
+      expectedTestDataList.map(_._2).toSet
+    )
+
+    testForEmptyTopics(fixture.kafkaEndpoints, groupId, topicInfoList)
+  }
+
+  "MockEventHandler" should "handle all events from multiple Kafka topics and then handle nil events" in { fixture =>
```
Comment (placccebo) [inline src/it/scala/com/bwsw/kafka/reader/KafkaReaderIntegrationTest.scala]: it should "process all events from multiple Kafka topics, save checkpoint data (indicating what events have been processed), run from a scratch and not receive any message"

**Review by placccebo — CHANGES_REQUESTED** [review]

File: src/it/scala/com/bwsw/kafka/reader/KafkaReaderIntegrationTest.scala (hunk 5 above)
Comment (MedvedevBW) [reply src/it/scala/com/bwsw/kafka/reader/KafkaReaderIntegrationTest.scala]: This is due to the rounding of an integer. Producer records is created for two topics in count of countOfTestData/2, so count of producer records will be countOfTestData/2 * 2. For example: countOfTestData = 11, count of records for first topic = 11/2 = 5, count of records for second topic = 11/2 = 5, so count for all topics = 11/2 * 2 = 10

File: src/it/scala/com/bwsw/kafka/reader/KafkaReaderIntegrationTest.scala (hunk 6 above)
Comment (MedvedevBW) [reply src/it/scala/com/bwsw/kafka/reader/KafkaReaderIntegrationTest.scala]: ok

File: src/it/scala/com/bwsw/kafka/reader/KafkaReaderIntegrationTest.scala (hunk 1 above)
Comment (MedvedevBW) [reply src/it/scala/com/bwsw/kafka/reader/KafkaReaderIntegrationTest.scala]: ok

File: src/it/scala/com/bwsw/kafka/reader/KafkaReaderIntegrationTest.scala (hunk 7 above)
Comment (MedvedevBW) [reply src/it/scala/com/bwsw/kafka/reader/KafkaReaderIntegrationTest.scala]: ok

File: src/it/scala/com/bwsw/kafka/reader/KafkaReaderIntegrationTest.scala (hunk 2 above)
Comment (MedvedevBW) [reply src/it/scala/com/bwsw/kafka/reader/KafkaReaderIntegrationTest.scala]: ok

File: src/it/scala/com/bwsw/kafka/reader/KafkaReaderIntegrationTest.scala (hunk 4 above)
Comment (MedvedevBW) [reply src/it/scala/com/bwsw/kafka/reader/KafkaReaderIntegrationTest.scala]: ok)

File: src/it/scala/com/bwsw/kafka/reader/KafkaReaderIntegrationTest.scala (hunk 3 above)
Comment (MedvedevBW) [reply src/it/scala/com/bwsw/kafka/reader/KafkaReaderIntegrationTest.scala]: fixed

**Review by placccebo — APPROVED** [review]

**Pull request merged**